### PR TITLE
Cohere support added

### PR DESCRIPTION
**Pull request by hasanraiyan**
#### **Summary**
This pull request introduces the integration of Cohere AI models into the project by adding API key management and model retrieval functionalities. The changes allow users to seamlessly connect with Cohere's services and utilize their suite of models.

#### **Changes Made**
1. **Environment Variables**  
   - Added support for the `COHERE_API_KEY` environment variable.  
   - Included instructions on obtaining the API key from Cohere's [dashboard](https://dashboard.cohere.com/api-keys).

2. **Cohere Model Management**  
   - Implemented a function `getCohereAIModel(apiKey: string, model: string)` to retrieve specified Cohere AI models using their API.
   - Integrated with `@ai-sdk/cohere`.

3. **Static Model Definitions**  
   - Defined a list of static models available through Cohere:
     - `Command R Plus Latest`
     - `Command R Latest`
     - `Command`
     - `c4AI Aya Expanse` (8B and 32B)  
     - Others as listed in the staticModels configuration.

4. **Configuration Enhancements**  
   - Modified case handling to include Cohere integration in the main execution path.
   - Added model mapping for a variety of Cohere's offerings.

#### **Instructions for Use**
1. Set the `COHERE_API_KEY` in your environment variables.
2. Use the updated configuration to specify a desired model from the Cohere static model list.
3. Follow the documentation for the `@ai-sdk/cohere` package for additional configuration options.

### CODE REVIEW

**Comment by wonderwhy-er** [comment]
Wo great job man with new commits.
Will need to test it bit later.
I like it!

**Comment by hasanraiyan** [comment]
> Wo great job man with new commits. Will need to test it bit later. I like it!

Hi @wonderwhy-er,

Thanks for the feedback! Let me know if you find anything during testing or have suggestions for improvement.

File: app/lib/.server/llm/stream-text.ts [hunk 1]
```diff
@@ -64,13 +63,20 @@ export function streamText(
...
 
-    return message; // No changes for non-user messages
+    return message; 
   });
 
+  const modelDetails = MODEL_LIST.find((m) => m.name === currentModel);
+
+  const dynamicMaxTokens =
+    modelDetails && modelDetails.maxTokenAllowed
+      ? Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)
```
Comment (wonderwhy-er) [inline app/lib/.server/llm/stream-text.ts:73]: So, I tested it and it works.
I am just wondering, why is this here?
Why are we not using just model tokens? Why min of both?

Comment (hasanraiyan) [reply app/lib/.server/llm/stream-text.ts:73]: Hi @wonderwhy-er,

Thanks for testing and sharing your feedback! Here's why I use `Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)`:

1. **Safety Check:** `MAX_TOKENS` acts as a safeguard to prevent issues if `modelDetails.maxTokenAllowed` is misconfigured or unexpectedly high.
2. **System Limits:** It ensures the token limit stays within the project's resource constraints, even if a model allows more tokens.
3. **Compatibility:** `MAX_TOKENS` was likely used in the code before, so keeping it avoids breaking any existing functionality.

If this adds unnecessary complexity, we can simplify the logic and rely only on `modelDetails.maxTokenAllowed`. Let me know what you think!

Comment (wonderwhy-er) [reply app/lib/.server/llm/stream-text.ts:73]: I would use it as default if model does not have its own but allow it to be bigger than MAX_TOKENS

So if model does not have its own maxTokenAllowed then MAX_TOKENS is used.

You have code like this

  const dynamicMaxTokens =
    modelDetails && modelDetails.maxTokenAllowed
      ? Math.min(MAX_TOKENS, modelDetails.maxTokenAllowed)
      : MAX_TOKENS;
      
      Change it to
      
        const dynamicMaxTokens =
    modelDetails && modelDetails.maxTokenAllowed
      ? modelDetails.maxTokenAllowed
      : MAX_TOKENS;

**Comment by wonderwhy-er** [comment]
Tested and works!

**Comment by wonderwhy-er** [comment]
Thank you for your contribution!
Is there any way in which you would like to be featured if youtuber pickups and shares your contribution?
